### PR TITLE
droidcam: init at 1.6

### PR DESCRIPTION
**Pull request by suhr**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lukegb** [comment]
This doesn't seem to build - https://github.com/NixOS/nixpkgs/pull/108346/checks?check_run_id=1640694049

**Comment by suhr** [comment]
Yes, that's why it's marked as a draft.

**Comment by lukegb** [comment]
D'oh, missed that.

File: pkgs/applications/video/droidcam/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    libjpeg_turbo
+    gtk3
+    alsaLib
+    speex
+    libusbmuxd
+    libappindicator-gtk3
+  ];
+ 
+  postPatch = ''
+    substituteInPlace linux/src/droidcam.c --replace "/opt/droidcam-icon.png" "$out/share/icons/hicolor/droidcam.png"
```
Comment (SuperSandro2000) [inline pkgs/applications/video/droidcam/default.nix:32]:
```suggestion
    substituteInPlace linux/src/droidcam.c \
      --replace "/opt/droidcam-icon.png" "$out/share/icons/hicolor/droidcam.png"
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/droidcam/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ 
+  postPatch = ''
+    substituteInPlace linux/src/droidcam.c \
+      --replace "/opt/droidcam-icon.png" "$out/share/icons/hicolor/droidcam.png"
+  '';
+ 
+  preBuild = ''
+    cd linux
+    makeFlagsArray+=("JPEG=$(pkg-config --libs --cflags libturbojpeg)")
+  '';
```
Comment (collares) [inline pkgs/applications/video/droidcam/default.nix:39]: Needs `makeFlagsArray+=("USBMUXD=$(pkg-config --libs --cflags libusbmuxd-2.0)")` too.

Comment (collares) [inline pkgs/applications/video/droidcam/default.nix:39]:
```suggestion
    makeFlagsArray+=("USBMUXD=$(pkg-config --libs --cflags libusbmuxd-2.0)")
  '';
```

File: pkgs/applications/video/droidcam/default.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    install -Dt $out/bin droidcam droidcam-cli
+    install -D icon2.png $out/share/icons/hicolor/droidcam.png
+ 
+    runHook postInstall
+  '';
+ 
+  meta = with stdenv.lib; {
+    description = "Linux client for DroidCam app";
+    homepage = "https://github.com/aramg/droidcam";
+    license = licenses.gpl2;
```
Comment (jtojnar) [inline pkgs/applications/video/droidcam/default.nix:54]: `gpl2` is unclear, use `gpl2Plus` or `gpl2Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

**Comment by suhr** [comment]
Now it compiles and runs.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108346` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>droidcam</li>
</ul>
</details>

**Pull request merged**